temporian/proto/core.proto

```diff
@@ -96,65 +100,61 @@ enum DType {
   STRING = 6;
 }
 
-// Schema of an event set
+// Connections between operators
 message Node {
   // Identifier of the node. Should be unique across all the nodes of the
   // graph.
   optional string id = 1;
 
-  // Identifier of the sampling of the node.
+  // Identifier of the sampling ref.
   optional string sampling_id = 2;
 
-  // Features in the node.
+  // Identifiers of the feature refs.
   repeated string feature_ids = 3;
 
   optional string name = 4;
 
   optional string creator_operator_id = 5;
-}
 
-// Schema of a feature
+  // Description of the data transiting in the node.
+  optional Schema schema = 6;
```

Comment:
might be wrong on this one, but can't schemas be shared between nodes and features? if so, shouldn't it be defined as reference here like the operator is?

Reply:
Yes, a schema can be shared (and is shared in memory) between some nodes (e.g. nodes that don't change any index, feature name, or set of selected features). However, we don't have the semantic of schema equality. So conceptually, this would not make a difference.

I think this solution is sightly simplest (with the potential cost of having slightly larger serialized models).

Reply:
Yes, a schema can be shared (and is shared in memory) between some nodes (e.g. nodes that don't change any index, feature name, or set of selected features). However, we don't have the semantic of schema equality. So conceptually, this would not make a difference.

I think this solution is sightly simplest (with the potential cost of having slightly larger serialized models).